ihatemoney/tests/tests.py

```diff
@@ -2224,5 +2252,24 @@ def test_bill_pay_each(self):
                 self.assertEqual(bill.pay_each(), pay_each_expected)
 
 
+class TestCurrencyConverter(unittest.TestCase):
+    converter = CurrencyConverter()
+    converter._internal_cache.set("rates", {"USD": 1, "EUR": 0.81149})
+
+    def test_only_one_instance(self):
+        one = id(CurrencyConverter())
+        two = id(CurrencyConverter())
+        self.assertEqual(one, two)
+
+    def test_get_currencies(self):
+        self.assertCountEqual(
+            self.converter.get_currencies(), [("USD", "USD"), ("EUR", "EUR")]
```

Comment:
I don't understand what get_currencies is supposed to return? (and why it returns this!)

Reply:
@almet , get_currencies returns an array tuple pair of the currencies. It is used for currency selection in form.py and whenever a user selects a currency. It is the map from the choice to the value:

Ie I am shown "USD" and the application registers my currency as  "USD".

![image](https://user-images.githubusercontent.com/52840419/76815625-0fa04980-67f6-11ea-9caa-237d3761f5d4.png)
![image](https://user-images.githubusercontent.com/52840419/76815645-1f1f9280-67f6-11ea-9625-5395daf52312.png)

Reply:
Alright! Thanks for the explanation. That's why the currencies appear two times. I find it kinda weird to tie the logic of the forms to the CurrencyConverter.

Due to the name, I was expecting `get_currencies` to return a list of currencies (maybe with their representation with the dollar sign etc) !

Maybe should we not tie the two together, and do the adaptation for the forms in the forms module instead (`choices = [(c, c) for c in currency_helper.get_currencies()]`) ? What do you think?

Reply:
I mean, that's kinda violating DRY, but alright. At this point I just want to land this. So I am happy to change that.

Reply:
Done. Please let me know there is anything else.